heat/core/random.py

```diff
@@ -451,30 +484,48 @@ def rand(
     split = stride_tricks.sanitize_axis(shape, split)
     device = devices.sanitize_device(device)
     comm = communication.sanitize_comm(comm)
-    balanced = True
-
-    # generate the random sequence
-    if dtype == types.float32:
-        x_0, x_1, lshape, lslice = __counter_sequence(shape, torch.int32, split, device, comm)
-        x_0, x_1 = __threefry32(x_0, x_1, seed=__seed)
 
-        # combine the values into one tensor and convert them to floats
-        values = __int32_to_float32(torch.stack([x_0, x_1], dim=1).flatten()[lslice]).reshape(
-            lshape
-        )
-    elif dtype == types.float64:
-        x_0, x_1, lshape, lslice = __counter_sequence(shape, torch.int64, split, device, comm)
-        x_0, x_1 = __threefry64(x_0, x_1, seed=__seed)
+    if __rng == "Threefry":
+        # use Threefry RNG (note: batchparallel RNG comes at the very end in only line only)
+        balanced = True
+        # generate the random sequence
+        if dtype == types.float32:
+            x_0, x_1, lshape, lslice = __counter_sequence(shape, torch.int32, split, device, comm)
+            x_0, x_1 = __threefry32(x_0, x_1, seed=__seed)
+
+            # combine the values into one tensor and convert them to floats
+            values = __int32_to_float32(torch.stack([x_0, x_1], dim=1).flatten()[lslice]).reshape(
+                lshape
+            )
+        elif dtype == types.float64:
+            x_0, x_1, lshape, lslice = __counter_sequence(shape, torch.int64, split, device, comm)
+            x_0, x_1 = __threefry64(x_0, x_1, seed=__seed)
+
+            # combine the values into one tensor and convert them to floats
+            values = __int64_to_float64(torch.stack([x_0, x_1], dim=1).flatten()[lslice]).reshape(
+                lshape
+            )
+        else:
+            # Unsupported type
+            raise ValueError(f"dtype is none of ht.float32 or ht.float64 but was {dtype}")
 
-        # combine the values into one tensor and convert them to floats
-        values = __int64_to_float64(torch.stack([x_0, x_1], dim=1).flatten()[lslice]).reshape(
-            lshape
-        )
+        return DNDarray(values, shape, dtype, split, device, comm, balanced)
     else:
-        # Unsupported type
-        raise ValueError(f"dtype is none of ht.float32 or ht.float64 but was {dtype}")
-
-    return DNDarray(values, shape, dtype, split, device, comm, balanced)
+        # use batchparallel RNG
+        x = factories.__factory(
+            shape if shape != () else (1,),
```

Comment:
```suggestion
            shape if len(shape) > 0 else (1,),
```
easier to understand